mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md

```diff
@@ -0,0 +1,177 @@
+
+![Resample3D module](/images/tutorials/thirdparty/pytorch_example2_1.png "Resample3D module").
+
+The coordinates in PyTorch are also a little different than in MeVisLab, therefore you have to rotate the image. Add an `OrthoSwapFlip` module and connect it to the `Resample3D` module. Change *View* to *Other* and set *Orientation* to *YXZ*. Also check *Flip horizontal*, *Flip vertical* and *Flip depth*. *Apply* your changes.
+
+![OrthoSwapFlip module](/images/tutorials/thirdparty/pytorch_example2_2.png "OrthoSwapFlip module").
+
+You can use the Output Inspector to see the changes on the images after applying the resample and a swap or flip.
+
+{{< imagegallery 3 "images/tutorials/thirdparty/" "Original" "Resample3D" "OrthoSwapFlip">}}
+
+Add an `OrthoView2D` to your network and save the *\*.mlab* file.
+
+![OrthoView2D module](/images/tutorials/thirdparty/pytorch_example2_3.png "OrthoView2D module").
+
+## Integrate PyTorch and scripting
+For integrating PyTorch and Python scripting, we need a `PythonImage` module. Add it to your workspace. Right-click {{< mousebutton "right" >}} on the `PythonImage` module and select {{< menuitem "Grouping" "Add to new Group...">}}. Right-click {{< mousebutton "right" >}} your new group and select {{< menuitem "Grouping" "Add to new Group...">}}. Name your new local macro *DemoAI*, select a directory for your project and leave all settings as default.
+
+Our new module does not provide an input or output. 
+
+![DemoAI local macro](/images/tutorials/thirdparty/pytorch_example2_4.png "DemoAI local macro").
+
+### Add Interfaces to your local macro
+Right-click {{< mousebutton "right" >}} the local macro and select {{< menuitem "Related Files" "DemoAI.script">}}. MATE opens showing the *\*.script* file of our module. Add an input *Field* of type *Image*, an output *Field* using the *internalName* of the output of our `PythonImage` and a *Trigger* to start the segmentation.
+
+You should also already add a Python file in the *Commands* section.
+
+{{< highlight filename="DemoAI.script" >}}
+```Stan
+Interface {
+  Inputs {
+    Field inputImage { type = Image }
+  }
+  Outputs {
+    Field outImage { internalName = PythonImage.output0 }
+  }
+  Parameters {
+    Field start { type = Trigger }
+  }
+}
+
+Commands {
+  source = $(LOCAL)/DemoAI.py
+}
+```
+{{</highlight>}}
+
+In MATE, right-click {{< mousebutton "right" >}} the Project Workspace and add a new file *DemoAI.py* to your project. The workspace now contains an empty Python file.
+
+![Project Workspace](/images/tutorials/thirdparty/pytorch_example2_5.png "Project Workspace").
+
+Change to MeVisLab IDE, right-click {{< mousebutton "right" >}} the local macro and select {{< menuitem "Reload Definition">}}. Your new input and output interfaces are now available and you can connect images to your module.
```

Comment:
I am not completely sure so please correct me if I am wrong, but aren't we talking about a single interface as
in- and output connectors were added to the interface of the module?